tests/base/src/test/java/org/wildfly/security/ssl/SSLAuthenticationTest.java

```diff
@@ -731,60 +725,93 @@ public void testClientSideOcsp() throws Exception {
                 .setClientMode(true)
                 .build().create();
 
-        performConnectionTest(serverContextGood, clientContext, true);
-        performConnectionTest(serverContextRevoked, clientContext, false);
+
+        testCommunication(serverContextGood, clientContext, "OU=Elytron,O=Elytron,C=UK,ST=Elytron,CN=ocspCheckedGood",
+                null, null);
+        try {
+            testCommunication(serverContextRevoked, clientContext, null, null, null);
+            Assert.fail("Expected SSLHandshakeException not thrown");
+        } catch (SSLHandshakeException expected) {
+            //expected
+        }
     }
 
-    private SecurityIdentity performConnectionTest(SSLContext serverContext, String clientUri, boolean expectValid) throws Exception {
+    private void performConnectionTest(SSLContext serverContext, String clientUri, boolean expectValid, String expectedServerPrincipal, String expectedClientPrincipal, String expectedCipherSuite) throws Throwable {
         System.setProperty("wildfly.config.url", SSLAuthenticationTest.class.getResource("wildfly-ssl-test-config-v1_1.xml").toExternalForm());
         AccessController.doPrivileged((PrivilegedAction<Integer>) () -> Security.insertProviderAt(WildFlyElytronPasswordProvider.getInstance(), 1));
 
         AuthenticationContext context = AuthenticationContext.getContextManager().get();
         AuthenticationContextConfigurationClient contextConfigurationClient = AccessController.doPrivileged(AuthenticationContextConfigurationClient.ACTION);
         SSLContext clientContext = contextConfigurationClient.getSSLContext(URI.create(clientUri), context);
 
-        return performConnectionTest(serverContext, clientContext, expectValid);
+        try {
+            testCommunication(serverContext, clientContext, expectedServerPrincipal, expectedClientPrincipal, expectedCipherSuite);
+            if(!expectValid) Assert.fail("Expected SSLHandshakeException not thrown");
+        } catch (SSLHandshakeException expected) {
+            //expected
+        } catch (SSLException e) {
+            if (e.getCause() instanceof SocketException) return; //expected
+            throw new IllegalStateException("Unexpected SSLException", e);
+        }
     }
 
-    private SecurityIdentity performConnectionTest(SSLContext serverContext, SSLContext clientContext, boolean expectValid) throws Exception {
-        SSLServerSocketFactory sslServerSocketFactory = serverContext.getServerSocketFactory();
-
-        SSLServerSocket sslServerSocket = (SSLServerSocket) sslServerSocketFactory.createServerSocket(1111, 10, InetAddress.getLoopbackAddress());
+    private void testCommunication(SSLContext serverContext, SSLContext clientContext, String expectedServerPrincipal, String expectedClientPrincipal, String expectedCipherSuite) throws Throwable {
```

Comment:
It doesn't look like anything is using the ```expectedCipherSuite``` parameter so it can be removed here.

Reply:
Removed, thanks!